mlir/test/Dialect/Linalg/roundtrip-linalg-named-ops.mlir

```diff
@@ -0,0 +1,49 @@
+// RUN: mlir-opt %s -linalg-generalize-named-ops | mlir-opt --linalg-specialize-generic-ops | FileCheck %s
```

Comment:
It is probably better to test the specialization by itself to avoid unnecessary coupling.

Reply:
Have added tests to test separately.
-  mlir/test/Dialect/Linalg/transform-op-specialize_matmul.mlir 
- mlir/test/Dialect/Linalg/specialize-generic-ops.mlir

I added 'round-trip' test based on my own previous 'learning' experience (e.g. in LLVM/MLIR) where beginner looks at tests to understand/experiment. The round-trip brings home the feature/tools/options and the concept:  named => generic => named.